httpclient/src/main/java/com/okta/sdk/impl/http/httpclient/HttpClientRequestExecutor.java

```diff
@@ -342,26 +408,37 @@ private boolean isRedirect(HttpResponse response) {
                 response.getHeaders("Location").length > 0;
     }
 
+    private URI getRedirectUri(HttpResponse httpResponse) {
+        Header[] locationHeaders = httpResponse.getHeaders("Location");
+        String location = locationHeaders[0].getValue();
+        log.debug("Redirecting to: {}", location);
+        return URI.create(location);
+    }
+
     /**
      * Exponential sleep on failed request to avoid flooding a service with
      * retries.
      *
      * @param retries           Current retry count.
-     * @param previousException Exception information for the previous attempt, if any.
      */
-    private void pauseExponentially(int retries, RestException previousException) {
-        long delay;
+    private void pauseBeforeRetry(int retries, HttpResponse httpResponse, long timeElapsed) throws RestException {
+        long delay = -1;
+        long timeElapsedLeft = maxElapsedMillis - timeElapsed;
         if (backoffStrategy != null) {
-            delay = this.backoffStrategy.getDelayMillis(retries);
-        } else {
-            long scaleFactor = 300;
-            if (previousException != null && isThrottlingException(previousException)) {
-                scaleFactor = 500 + random.nextInt(100);
+            delay = Math.min(this.backoffStrategy.getDelayMillis(retries), timeElapsedLeft);
+        } else if (httpResponse != null && httpResponse.getStatusLine().getStatusCode() == 429) {
+            delay = get429DelayMillis(httpResponse);
+            if (!shouldRetry(retries, timeElapsed + delay)) {
+                throw new RestException("HTTP 429: Too Many Requests.  Exceeded request rate limit in the allotted amount of time.");
```

Comment:
Do we need to add this error text?  Can we not pass through what we get on the response?

Reply:
It does, via this: https://github.com/okta/okta-sdk-java/pull/197/files#diff-51ae4451c073ee1f09052036df605180R309

Reply:
Though thinking about it again, the message could be better 🤔